### PR TITLE
pkg: clarify that current `download` target is only for git

### PR DESCRIPTION
**Pull request by authmillenon**
Enhancement to #5138. I did not provide a general download method for HTTP since I think it is too diverse. We do not have other sources than git or HTTP yet so I did not provide e.g. SVN either ;-).

### CODE REVIEW

**Comment by kaspar030** [comment]
Hm. The git template was not deprecated. 

And don't you think we can devise a download target that can distinguish between git and http downloads:

- PKG_URL_FORMAT can always override
- url ends with "git" -> git
- url ends with zip, tbz2, tgz, tar.gz, ... -> http

something like that?

File: pkg/pkg.mk [hunk 1]
```diff
@@ -4,15 +4,17 @@
 PKG_DIR?=$(CURDIR)
 PKG_BUILDDIR?=$(BINDIR)/pkg/$(PKG_NAME)
 
-download: $(PKG_BUILDDIR)/.downloaded
```
Comment (kaspar030) [inline pkg/pkg.mk]: why phony?

Comment (authmillenon) [inline pkg/pkg.mk]: Because `git-download` must be executed because it has (and should not have) no file equivalent.

**Comment by authmillenon** [comment]
> And don't you think we can devise a download target that can distinguish between git and http downloads

Can we? Probably yes, but why not KISS let the packet decide what download target to use. Since it needs to provide a URL and version, too this makes the most sense to me.

> PKG_URL_FORMAT can always override
> * url ends with "git" -> git

This is a convention, not a fixed thing and any repo could just not do it making things harder for a package maintainer (try it: you can clone your local RIOT repository into another directory using `git clone`).

> * url ends with zip, tbz2, tgz, tar.gz, ... -> http

What if the file is provided over FTP, not HTTP?

**Comment by authmillenon** [comment]
> Hm. The git template was not deprecated.

args, did not see that you changed that too. Will revert and adapt.

**Comment by authmillenon** [comment]
Done.

File: pkg/pkg.mk (hunk 1 above)
Comment (kaspar030) [inline pkg/pkg.mk]: totally correct.

**Comment by kaspar030** [comment]
> url ends with zip, tbz2, tgz, tar.gz, ... -> http
> 
> What if the file is provided over FTP, not HTTP?

Does curl or wget care?

**Comment by kaspar030** [comment]
> And don't you think we can devise a download target that can distinguish between git and http downloads
> 
> Can we? Probably yes, but why not KISS and let the packet decide what download target to use. Since it needs to provide a URL and version, too this makes the most sense to me.
> 
>     PKG_URL_FORMAT can always override

I think we can guess from sane defaults, but allow overriding. OpenWrt does it the same. But I don't have strong feelings here.

**Comment by kaspar030** [comment]
ACK, pls squash!

**Comment by authmillenon** [comment]
Done.

**Comment by kaspar030** [comment]
&go!

**Pull request merged**